addons/project/models/res_partner.py

```diff
@@ -61,3 +61,34 @@ def _create_portal_users(self):
                 'active': True,
             })
         return created_users
+
+    def action_view_tasks(self):
+        self.ensure_one()
+        action = self.env["ir.actions.actions"]._for_xml_id("project.action_view_all_task")
+        action['display_name'] = _("%(partner_name)s's Tasks", partner_name=self.name)
+        action['context'] = {
+            'default_partner_id': self.id,
+        }
+        views = []
+        for view in action['views']:
+            if view[1] == 'form':
+                views.append((self.env.ref('project.view_task_form_res_partner').id, view[1]))
+            elif view[1] == 'kanban':
+                views.append((self.env.ref('project.view_task_kanban_res_partner').id, view[1]))
+            else:
+                views.append(view)
+        action['views'] = views
```

Comment:
If you only need the id: `self.env['ir.model.data']._xmlid_to_res_id(...)`.
Also, how about this? :neckbeard: 
```suggestion
        action['views'] = [
            (self.env['ir.model.data']._xmlid_to_res_id(f'project.view_task_{name}_res_partner'), name)
            if name in ('form', 'kanban') else
            (res_id, name)
            for res_id, name in action['views']
        ]
```

Reply:
Also, I would rather have that in the "else" block, as you override it in the "if" block.

Reply:
Or `view_id`, `view_type`, just like you did on enterprise :ok_hand: